Error when using an array of a struct in an intreface

**V version: V 0.2.2 c1d4074** 
**OS: Manjaro Linux** 

**What did you do?**
I tried running this code
```
struct Room {}


interface House {
    rooms []Room 
    
}
```

**What did you expect to see?**
no error
 
**What did you see instead?**
this error
```
/tmp/v/main.1947781590065766840.tmp.c:719: error: ';' expected (got "Array_main__Room")
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -791,9 +791,6 @@ pub fn (mut g Gen) write_typedef_types() {
 					}
 				}
 			}
-			.interface_ {
-				g.write_interface_typesymbol_declaration(typ)
-			}
 			.chan {
 				if typ.name != 'chan' {
 					g.type_definitions.writeln('typedef chan $typ.cname;')
@@ -823,6 +820,20 @@ static inline void __${typ.cname}_pushval($typ.cname ch, $el_stype val) {
 			else {
 				continue
 			}
+		}
+	}
+
+	// Generating interfaces after all the common types have been defined
+	// to prevent generating interface struct before definition of field types
+	for typ in g.table.type_symbols {
+		if typ.name in c.builtins {
+			continue
+		}
+		match typ.kind {
+			.interface_ {
+				g.write_interface_typesymbol_declaration(typ)
+			}
+			else {}
 		}
 	}
 }

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -826,14 +826,8 @@ static inline void __${typ.cname}_pushval($typ.cname ch, $el_stype val) {
 	// Generating interfaces after all the common types have been defined
 	// to prevent generating interface struct before definition of field types
 	for typ in g.table.type_symbols {
-		if typ.name in c.builtins {
-			continue
-		}
-		match typ.kind {
-			.interface_ {
-				g.write_interface_typesymbol_declaration(typ)
-			}
-			else {}
+		if typ.kind == .interface_ && typ.name !in c.builtins {
+			g.write_interface_typesymbol_declaration(typ)
 		}
 	}
 }

--- a/vlib/v/tests/interface_fields_typearray_test.v
+++ b/vlib/v/tests/interface_fields_typearray_test.v
@@ -1,0 +1,7 @@
+struct Test {}
+
+interface Testing {
+	tests []Test
+}
+
+fn test_field_typearray() {}

